### ISSUE TITLE
Failed to insert 'vboxpci' module on my host with kernel 4.14

### ISSUE THREAD

**Issue by kouros17** [manjaro/packages-core#86]
In every **kernel 4.14** version I get a boot error message with **systemd-modules-load service** and more particularly with **vboxpci** module. 

```
╰─ systemctl status systemd-modules-load
● systemd-modules-load.service - Load Kernel Modules
   Loaded: loaded (/usr/lib/systemd/system/systemd-modules-load.service; static; vendor pres
   Active: failed (Result: exit-code) since Tue 2017-10-17 12:24:11 EEST; 2h 6min ago
     Docs: man:systemd-modules-load.service(8)
           man:modules-load.d(5)
  Process: 226 ExecStart=/usr/lib/systemd/systemd-modules-load (code=exited, status=1/FAILUR
 Main PID: 226 (code=exited, status=1/FAILURE)

Οκτ 17 12:24:10 manjaro systemd-modules-load[226]: Inserted module 'sg'
Οκτ 17 12:24:10 manjaro systemd-modules-load[226]: Inserted module 'vboxdrv'
Οκτ 17 12:24:10 manjaro systemd-modules-load[226]: Inserted module 'vboxnetadp'
Οκτ 17 12:24:10 manjaro systemd-modules-load[226]: Inserted module 'vboxnetflt'
Οκτ 17 12:24:11 manjaro systemd-modules-load[226]: Failed to insert 'vboxpci': No such file 
Οκτ 17 12:24:11 manjaro systemd-modules-load[226]: Failed to insert 'vboxpci': No such file 
Οκτ 17 12:24:11 manjaro systemd-modules-load[226]: Failed to insert 'vboxpci': No such file 
Οκτ 17 12:24:11 manjaro systemd[1]: systemd-modules-load.service: Main process exited, code=
Οκτ 17 12:24:11 manjaro systemd[1]: systemd-modules-load.service: Failed with result 'exit-c
Οκτ 17 12:24:11 manjaro systemd[1]: Failed to start Load Kernel Modules.
```
And:

```
╰─ journalctl _PID=226
manjaro systemd-modules-load[226]: Failed to insert 'vboxpci': No such file or directory
```

**Comment by philmmanjaro** [comment]
Well, it is still provided by **dkms** file: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/virtualbox&id=ff2fb7a6beb99b78c26a658ffee51899b43d7b09#n277

And still provided by **linux414-virtualbox-host-modules**:

```
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/vboxpci.ko.gz
```

**Comment by philmmanjaro** [comment]
Also **dkms** still don't compile them with given reasons ...

```
==> dkms install vboxguest/5.1.30_OSE -k 4.14.0-1-MANJARO
Error! Bad return status for module build on kernel: 4.14.0-1-MANJARO (x86_64)
Consult /var/lib/dkms/vboxguest/5.1.30_OSE/build/make.log for more information.
==> No kernel 4.13.0-1.170718-MANJARO-SX modules. You must install them to use DKMS!
==> dkms install vboxhost/5.1.30_OSE -k 4.14.0-1-MANJARO
Error! Bad return status for module build on kernel: 4.14.0-1-MANJARO (x86_64)
Consult /var/lib/dkms/vboxhost/5.1.30_OSE/build/make.log for more information.

```

**Comment by philmmanjaro** [comment]
Needed patches can be always found here: https://github.com/manjaro/packages-extra/tree/master/linux414-extramodules/virtualbox-modules

**Comment by philmmanjaro** [comment]
No issue. Please use the binary packages:

```
==> Erstelle Paket: linux414-virtualbox-modules 5.1.30-0.10 (Di 17. Okt 11:27:04 EDT 2017)
==> Prüfe Laufzeit-Abhängigkeiten...
==> Prüfe Buildtime-Abhängigkeiten...
==> Empfange Quellen...
  -> linux-4.14.patch gefunden
  -> linux-4.14-host.patch gefunden
==> Überprüfe source Dateien mit sha256sums...
    linux-4.14.patch ... Durchgelaufen
    linux-4.14-host.patch ... Durchgelaufen
==> Entpacke Quellen...
==> Beginne prepare()...
patching file vboxvideo/vbox_drv.c
patching file vboxvideo/vbox_mode.c
patching file vboxvideo/vbox_fb.c
patching file vboxnetflt/linux/VBoxNetFlt-linux.c
==> Beginne build()...
  -> Host modules

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j16 KERNELRELEASE=4.14.0-1-MANJARO -C /usr/lib/modules/4.14.0-1-MANJARO/build M=/home/phil/dev/git/manjaro/repositories/stable-repos/extra/linux414-extramodules/virtualbox-modules/src/dkms/vboxhost/5.1.30_OSE/build.....
cleaning build area...
Kernel cleanup unnecessary for this kernel.  Skipping...

DKMS: build completed.
  -> Guest modules

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j16 KERNELRELEASE=4.14.0-1-MANJARO -C /usr/lib/modules/4.14.0-1-MANJARO/build M=/home/phil/dev/git/manjaro/repositories/stable-repos/extra/linux414-extramodules/virtualbox-modules/src/dkms/vboxguest/5.1.30_OSE/build.....
cleaning build area...
Kernel cleanup unnecessary for this kernel.  Skipping...

DKMS: build completed.
==> Betrete fakeroot Umgebung...
==> Beginne package_linux414-virtualbox-host-modules()...
==> Säubere Installation...
  -> Entferne libtool Dateien...
  -> Bereinige ungewollte Dateien...
  -> Entferne statische Bibliotheken...
  -> Entferne unnötige Symbole aus Binär-Dateien und Bibliotheken...
  -> Komprimiere Man-Pages und Info-Seiten...
==> Prüfe auf Paketierungsprobleme...
==> Erstelle Paket "linux414-virtualbox-host-modules"...
  -> Erstelle .PKGINFO Datei...
  -> Erstelle .BUILDINFO Datei...
  -> Füge install Datei hinzu...
  -> Erstelle .MTREE-Datei...
  -> Komprimiere Paket... 
==> Beginne package_linux414-virtualbox-guest-modules()...
==> Säubere Installation...
  -> Entferne libtool Dateien...
  -> Bereinige ungewollte Dateien...
  -> Entferne statische Bibliotheken...
  -> Entferne unnötige Symbole aus Binär-Dateien und Bibliotheken...
  -> Komprimiere Man-Pages und Info-Seiten...
==> Prüfe auf Paketierungsprobleme...
==> Erstelle Paket "linux414-virtualbox-guest-modules"...
  -> Erstelle .PKGINFO Datei...
  -> Erstelle .BUILDINFO Datei...
  -> Füge install Datei hinzu...
  -> Erstelle .MTREE-Datei...
  -> Komprimiere Paket... 
==> Verlasse fakeroot Umgebung.
==> Beendete Erstellung: linux414-virtualbox-modules 5.1.30-0.10 (Di 17. Okt 11:27:29 EDT 2017)
```

**Comment by kouros17** [comment]
I tried to build them with dkms and makepkg but I got the same errors like you.
How can I use the binary packages?

**Comment by philmmanjaro** [comment]
By installing them ...

**Comment by kouros17** [comment]
I have already installed **linux414-virtualbox-host-modules  5.1.30-0.10** on my host. 
What else should I install?

**Comment by philmmanjaro** [comment]
Normally that is all to do. You can check with `pacman -Ql linux414-virtualbox-host-modules` the modules.

**Comment by kouros17** [comment]
Seems all the modules (and **vboxpci**) are there like 4.13 kernel:
```
╰─ pacman -Ql linux414-virtualbox-host-modules
linux414-virtualbox-host-modules /etc/
linux414-virtualbox-host-modules /etc/modules-load.d/
linux414-virtualbox-host-modules /etc/modules-load.d/linux414-virtualbox-host-modules.conf
linux414-virtualbox-host-modules /usr/
linux414-virtualbox-host-modules /usr/lib/
linux414-virtualbox-host-modules /usr/lib/modules/
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/vboxdrv.ko.gz
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/vboxnetadp.ko.gz
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/vboxnetflt.ko.gz
linux414-virtualbox-host-modules /usr/lib/modules/extramodules-4.14-MANJARO/vboxpci.ko.gz
```
And:
```
╰─ pacman -Ql linux413-virtualbox-host-modules
linux413-virtualbox-host-modules /etc/
linux413-virtualbox-host-modules /etc/modules-load.d/
linux413-virtualbox-host-modules /etc/modules-load.d/linux413-virtualbox-host-modules.conf
linux413-virtualbox-host-modules /usr/
linux413-virtualbox-host-modules /usr/lib/
linux413-virtualbox-host-modules /usr/lib/modules/
linux413-virtualbox-host-modules /usr/lib/modules/extramodules-4.13-MANJARO/
linux413-virtualbox-host-modules /usr/lib/modules/extramodules-4.13-MANJARO/vboxdrv.ko.gz
linux413-virtualbox-host-modules /usr/lib/modules/extramodules-4.13-MANJARO/vboxnetadp.ko.gz
linux413-virtualbox-host-modules /usr/lib/modules/extramodules-4.13-MANJARO/vboxnetflt.ko.gz
linux413-virtualbox-host-modules /usr/lib/modules/extramodules-4.13-MANJARO/vboxpci.ko.gz
```
I don't understand why "**Failed to insert 'vboxpci': No such file or directory**" on **4.14** kernel but not on **4.13** kernel.

**Comment by philmmanjaro** [comment]
I only made it to compile. I never tested if it acutally loads. What happens with `modprobe vboxpci`?

**Comment by kouros17** [comment]
╰─ sudo modprobe vboxpci
modprobe: ERROR: could not insert 'vboxpci': Unknown symbol in module, or unknown parameter (see dmesg)

**Comment by oberon2007** [comment]
Seems strange. I am running vbox on linux414 all the time without problem here.

**Comment by philmmanjaro** [comment]
@kouros17: try to recompile the kernel with my patches by downloading the extramodules folder from **packages-extra** git repo.

**Comment by kouros17** [comment]
O.k I'll try...I found this:
https://bugs.archlinux.org/task/54252

**Comment by oberon2007** [comment]
are you using spl extramodule? https://github.com/zfsonlinux/spl/issues/656

**Comment by kouros17** [comment]
@oberon2007 No I don't.

**Comment by kouros17** [comment]
@philmmanjaro when I try to recompile the kernel extramodules with patches I get this to the end:
```
:: Running post-transaction hooks...
(1/2) Install DKMS modules
==> dkms install vboxhost/5.1.30_OSE -k 4.13.7-1-MANJARO
==> dkms install vboxguest/5.1.30_OSE -k 4.14.0-1-MANJARO
Error! Bad return status for module build on kernel: 4.14.0-1-MANJARO (x86_64)
Consult /var/lib/dkms/vboxguest/5.1.30_OSE/build/make.log for more information.
==> dkms install vboxguest/5.1.30_OSE -k 4.13.7-1-MANJARO
==> dkms install vboxhost/5.1.30_OSE -k 4.14.0-1-MANJARO
Error! Bad return status for module build on kernel: 4.14.0-1-MANJARO (x86_64)
Consult /var/lib/dkms/vboxhost/5.1.30_OSE/build/make.log for more information.
(2/2) Arming ConditionNeedsUpdate...
==> Retrieving sources...
  -> Found linux-4.14.patch
  -> Found linux-4.14-host.patch
==> Validating source files with sha256sums...
    linux-4.14.patch ... FAILED
    linux-4.14-host.patch ... FAILED
==> ERROR: One or more files did not pass the validity check!
```

**Comment by philmmanjaro** [comment]
@kouros17: did you do a git clone? Else check if the patches are sane and do a **updpkgsums** before you try to compile.

**Comment by kouros17** [comment]
When I try to do git clone I get:
```
╰─ git clone https://github.com/manjaro/packages-extra/tree/master/linux414-extramodules/virtualbox-modules
Cloning into 'virtualbox-modules'...
remote: Not Found
fatal: repository 'https://github.com/manjaro/packages-extra/tree/master/linux414-extramodules/virtualbox-modules/' not found
```
So I downloaded manually the 5 files of virtualbox-modules folder and I tried to recompile them with **makepkg**. This time I did **updpkgsums** before the compile, the process went further,  but in the end I got:
```
==> Starting build()...
  -> Host modules

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j8 KERNELRELEASE=4.14.0-1-MANJARO -C /usr/lib/modules/4.14.0-1-MANJARO/build M=/home/giorgos/Builds/virtualbox-modules/src/dkms/vboxhost/5.1.30_OSE/build.....
cleaning build area...
Kernel cleanup unnecessary for this kernel.  Skipping...

DKMS: build completed.
  -> Guest modules

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j8 KERNELRELEASE=4.14.0-1-MANJARO -C /usr/lib/modules/4.14.0-1-MANJARO/build M=/home/giorgos/Builds/virtualbox-modules/src/dkms/vboxguest/5.1.30_OSE/build.....(bad exit status: 2)
Error! Bad return status for module build on kernel: 4.14.0-1-MANJARO (x86_64)
Consult /home/giorgos/Builds/virtualbox-modules/src/dkms/vboxguest/5.1.30_OSE/build/make.log for more information.
==> ERROR: A failure occurred in build().
    Aborting...
```

**Comment by kouros17** [comment]
I did updpkgsums before and this time the compilation completed but I still get the same boot error message.

**Comment by kouros17** [comment]
Eventually resolved with **4.14.r1028.g11224e1-1**

**Comment by jonathonf** [comment]
Just linking in the associated [forum thread](https://forum.manjaro.org/t/unstable-update-2017-10-27-kde-gnome-deepin-firefox-mesa/33561/16?u=jonathon).
